pages/agent/v3/elastic_ci_aws/troubleshooting.md

```diff
@@ -35,6 +35,12 @@ Resource shortage can cause this issue. See the Auto Scaling group's Activity lo
 
 To fix this issue, change or add more instance types to the `InstanceTypes` template parameter. If 100% of your existing instances are Spot Instances, switch some of them to On-Demand Instances by setting `OnDemandPercentage` parameter to a value above zero.
 
+## Instances are abruptly terminated
+
+This can happen when using Spot Instances. AWS EC2 sends a notification to a spot instance 2 minutes prior to termination. The agent intercepts that notification and attempts to gracefully shut down. If the instance does not shut down gracefully in that time, it is terminated.
```

Comment:
@james2791 do you have an example of the log line we can document for people to use to know if their instance has been reclaimed?

Reply:
Some ideas for improvements:
1. Document adding retries to the step so it automatically retries if reclaimed.
2. Document how to search the logs to know if an instance has been reclaimed.

Reply:
Example: `| 2023-07-31 19:19:23.432 | level=info msg="Received termination notice" instanceId=i-abcd notice=spot | i-abcd | 444793955923:/buildkite/lifecycled |`